apps/clock/js/alarm.js

```diff
@@ -175,6 +250,56 @@ define(function(require, exports, module) {
           this._notifyChanged(/* removed = */ true);
         });
       });
+    },
+
+    /** Sort all the alarms and return the next upcoming
+     *  alarm from the list
+     */
+    sortAlarms: function(storeAlarms) {
+      storeAlarms.sort(function(a,b) {
+        return new Date('1970/01/01 '+a.hour+':'+a.minute)
+          - new Date('1970/01/01 ' + b.hour+':'+b.minute);
```

Comment:
Is there a reason not to just compare `hour` and `minute` directly? Constructing a Date in this way seems unnecessary.